build.sc

```diff
@@ -1559,191 +1650,301 @@ object dev extends MillModule {
   }
 }
 
+
+/** Generates the mill documentation with Antora. */
 object docs extends Module {
+  // This module isn't really a ScalaModule, but we use it to generate
+  // consolidated documentation using the Scaladoc tool.
+  object site extends UnidocModule {
+    def scalaVersion = Deps.scalaVersion
 
-  /** Generates the mill documentation with Antora. */
-  object antora extends Module {
-    private val npmExe = if (scala.util.Properties.isWin) "npm.cmd" else "npm"
-    private val antoraExe = if (scala.util.Properties.isWin) "antora.cmd" else "antora"
-    def npmBase: T[os.Path] = T.persistent { T.dest }
-    def prepareAntora(npmDir: os.Path) = {
-      Jvm.runSubprocess(
-        commandArgs = Seq(
-          npmExe,
-          "install",
-          "@antora/cli@3.0.1",
-          "@antora/site-generator-default@3.0.1",
-          "gitlab:antora/xref-validator",
-          "@antora/lunr-extension@v1.0.0-alpha.6"
-        ),
-        envArgs = Map(),
-        workingDir = npmDir
-      )
-    }
-    def runAntora(npmDir: os.Path, workDir: os.Path, args: Seq[String])(implicit
-        ctx: mill.api.Ctx.Log
-    ) = {
-      prepareAntora(npmDir)
-      val cmdArgs =
-        Seq(s"${npmDir}/node_modules/.bin/${antoraExe}") ++ args
-      ctx.log.debug(s"command: ${cmdArgs.mkString("'", "' '", "'")}")
-      Jvm.runSubprocess(
-        commandArgs = cmdArgs,
-        envArgs = Map("CI" -> "true"),
-        workingDir = workDir
-      )
-      PathRef(workDir / "build" / "site")
+    def moduleDeps = build.millInternal.modules.collect { case m: MillApiModule => m}
+
+    def unidocSourceUrl = T{
+      val sha = VcsVersion.vcsState().currentRevision
+      Some(s"https://github.com/com-lihaoyi/mill/blob/$sha")
     }
-    def source: Source = T.source(millSourcePath)
-    def supplementalFiles = T.source(millSourcePath / "supplemental-ui")
-    def devAntoraSources: Target[PathRef] = T {
-      val dest = T.dest
-      shared.mycopy(source().path, dest, mergeFolders = true)
-      val lines = os.read(dest / "antora.yml").linesIterator.map {
-        case l if l.startsWith("version:") =>
-          s"version: 'master'" + "\n" + s"display-version: '${millVersion()}'"
-        case l if l.startsWith("    mill-version:") =>
-          s"    mill-version: '${millVersion()}'"
-        case l if l.startsWith("    mill-last-tag:") =>
-          s"    mill-last-tag: '${millLastTag()}'"
-        case l => l
+  }
+
+  /**
+   * Mix this in to any [[ScalaModule]] to provide a [[unidocSite]] target that
+   * can be used to build a unified scaladoc site for this module and all of
+   * its transitive dependencies
+   */
+  trait UnidocModule extends ScalaModule {
```

Comment:
This gets deleted in `ci/patch-mill-bootstrap.sh`, and instead uses the version published in `mill.scalalib`